EXILED/Exiled.Events/Patches/Events/Player/Shooting.cs

```diff
@@ -39,34 +34,59 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
             List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
 
             Label returnLabel = generator.DefineLabel();
-
-            int offset = 1;
-            int index = newInstructions.FindIndex(instruction => instruction.opcode == OpCodes.Stloc_2) + offset;
+            Label continueLabel = generator.DefineLabel();
+
+            /*
+            [] <= Here
+            IL_0078: ldarg.2      // processingMethod
+            IL_0079: ldarg.0      // this
+            IL_007a: ldfld        class ReferenceHub InventorySystem.Items.Firearms.Modules.Misc.ShotBacktrackData::PrimaryTargetHub
+            IL_007f: callvirt     instance void class [mscorlib]System.Action`1<class ReferenceHub>::Invoke(!0/*class ReferenceHub* /)
+             */
+            int hasTargetIndex = newInstructions.FindIndex(instruction => instruction.IsLdarg(2));
+            List<Label> hasTargetLabels = newInstructions[hasTargetIndex].ExtractLabels();
+
+            /*
+            [] <= Here
+            IL_0092: ldarg.2      // processingMethod
+            IL_0093: ldnull       // null
+            IL_0094: callvirt     instance void class [mscorlib]System.Action`1<class ReferenceHub>::Invoke(!0/*class ReferenceHub* /)
+             */
+            int noTargetIndex = newInstructions.FindIndex(hasTargetIndex + 1, instruction => instruction.IsLdarg(2));
+            List<Label> noTargetLabels = newInstructions[noTargetIndex].ExtractLabels();
```

Comment:
There is no point at `.FindIndex(hasTargetIndex + 1...`, you can use `FindLastIndex` for that
```suggestion
            int noTargetIndex = newInstructions.FindLastIndex(instruction => instruction.IsLdarg(2));
            List<Label> noTargetLabels = newInstructions[noTargetIndex].ExtractLabels();
```

Reply:
the options are exactly the same in this case in terms of logic or efficiency, personally for me it is important that LastIndex does not take extra arguments and looks smaller🥹

Reply:
> FindIndex is more logically correct because I am looking specifically for the second Ldarg.2 instruction, it being last is a coincidence

well, there are only 2 calls, when there is a primary target, and when there is no primary target. personally for me there is no difference in these options, except that LastIndex is less saturated, but if you think otherwise then okk

Reply:
*but still I think the most important parameter of a patch is readability and understandability, you should not neglect it. The main thing in transpilers is not to write overly hard smart things, so that then everyone would struggle to figure out what you meant*

This case isnt like that ofc, but still😈